### PR TITLE
aws: tag instances on creation, add 'createdby' tag

### PR DESCRIPTION
**Pull request by euank**
This will make cleaning up aws instances reliably easier.

See the individual commits. If desired, I can break out the second one to a separate PR since these are only tangentially related.

### CODE REVIEW

**Comment by euank** [comment]
Updated to just be a hardcoded "kola" tag value.

**Comment by arithx** [comment]
Do we want this still split into two commits one of which adds the `name` and the other which adds the `createdBy`?

File: platform/api/aws/ec2.go [hunk 1]
```diff
@@ -85,6 +85,10 @@ func (a *API) CreateInstances(name, keyname, userdata string, count uint64) ([]*
 						Key:   aws.String("Name"),
 						Value: aws.String(name),
 					},
+					&ec2.Tag{
+						Key:   aws.String("CreatedBy"),
+						Value: aws.String("kola"),
```
Comment (bgilbert) [inline platform/api/aws/ec2.go]: This is platform code, so should be `mantle`.

**Comment by euank** [comment]
Agreed that I'd prefer them as separate commits since they are distinct changes.. and you can always operate on the merge commit regardless of whether I squash them or not.

**Pull request merged**